microcosm_pubsub/envelope.py

```diff
@@ -210,6 +210,36 @@ def parse_body(self, raw_message):
         return raw_message
 
 
+class LambdaSQSEnvelope(SNSMessageBodyParser, CodecMediaTypeAndContentParser, SQSEnvelope):
+    """
+    Enveloping strategy for AWS Lambda messages
+    Same as Codec, but Lambda receives message in little bit different format.
+    """
+    def parse_message_id(self, raw_message):
+        return raw_message["messageId"]
+
+    def parse_receipt_handle(self, raw_message):
+        return raw_message["receiptHandle"]
+
+    def parse_body(self, raw_message):
+        return raw_message["body"]
+
+    def validate_md5(self, raw_message, body):
+        """
+        Validate the message body.
+
+        Just checks for tampering; schema validation occurs once we know the type of message.
+
+        """
+        expected_md5_of_body = raw_message["md5OfBody"]
+        actual_md5_of_body = md5(body).hexdigest()
+        if expected_md5_of_body != actual_md5_of_body:
+            raise Exception("MD5 validation failed. Expected: {} Actual: {}".format(
```

Comment:
A more specific exception class would be good, potentially a `ValidationError` as done in `codecs.py`

Reply:
Ok, sure. I took this code as-is from SQSEnvelope base class.